### PR TITLE
Improve test for easycrypt extraction

### PR DESCRIPTION
**Pull request by cassiersg**
Split out from #846

### CODE REVIEW

File: compiler/scripts/extract-and-check [hunk 1]
```diff
@@ -2,24 +2,28 @@
...
-trap "rm -r ${DIR}" EXIT
+# trap "rm -r ${DIR}" EXIT
+
+ECLIB=$(dirname $0)/../../eclib
 
 set -x
 
-$(dirname $0)/../jasminc ${ARCH} -oecarray ${DIR} -oec ${EC} "$@"
-$(dirname $0)/../jasminc ${ARCH} -oecarray ${DIR} -CT -oec ${CT} "$@"
+echo $(dirname $0)/../jasminc ${ARCH} -oecarray ${DIR} -oec ${EC} "$2"
```
Comment (vbgl) [inline compiler/scripts/extract-and-check:23]: '$2' seems wrong here.

CI says: `Error: File  not found.`

Comment (cassiersg) [reply compiler/scripts/extract-and-check:25]: The issue was coming from the `shift` above, from an incorrect merge.

**Comment by cassiersg** [comment]
@vbgl The issue should be fixed.

**Comment by vbgl** [comment]
Well, there seems to be exactly the same issue with the `extract-golden` file.

**Comment by vbgl** [comment]
> The fixes to the testing infrastructure should be moved to a different PR (and merged quickly).

This applies to this PR too…

The first two commits are OK. The third one is a new feature. The last one is a draft for a new feature mixed up with a fix for an unconfirmed issue (please file a bug report if you think there is a real problem).

**Comment by cassiersg** [comment]
I left only the two first commits - the rest will follow later when I have time to refactor the commits.

BTW, is there any test to ensure that the extraction is correct, beyond being accepted by easycrypt ?

**Review by vbgl — APPROVED** [review]

**Pull request merged**